classes/ProductFilter/FilterApplication/AttributeQueryBuilder/FeatureQueryBuilder.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License version 3.0
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License version 3.0
+ */
+
+namespace PrestaShop\Module\PsxMarketingWithGoogle\ProductFilter\FilterApplication\AttributeQueryBuilder;
+
+use Context;
+use DbQuery;
+use PrestaShop\Module\PsxMarketingWithGoogle\ProductFilter\Condition;
+use PrestaShop\Module\PsxMarketingWithGoogle\Repository\LanguageRepository;
+
+class FeatureQueryBuilder implements QueryBuilderInterface
+{
+    /**
+     * @var Context
+     */
+    protected $context;
+
+    /**
+     * @var LanguageRepository
+     */
+    protected $languageRepository;
+
+    /**
+     * @var string
+     */
+    protected $currentLanguageIsoCode;
+
+    public function __construct(
+        Context $context,
+        LanguageRepository $languageRepository
+    ) {
+        $this->context = $context;
+        $this->languageRepository = $languageRepository;
+        $this->currentLanguageIsoCode = $this->languageRepository->getIsoById(
+            (int) $this->context->language->id
+        );
+    }
+
+    public function addWhereFromFilter(DbQuery $query, $filter): DbQuery
+    {
+        $uniqueFeature = [];
+
+        foreach ($filter['values'] as $value) {
+            if (!isset($uniqueFeature[$value['id']])) {
+                $uniqueFeature[$value['id']] = $value;
+            }
+        }
```

Comment:
without the isset, the value would be overridden.
Would it be a problem ?
That would mean that the last duplicate would be the one taken.
```php
        foreach ($filter['values'] as $value) {
              $uniqueFeature[$value['id']] = $value;
        }
```